lsst/ts/laser/csc.py

```diff
@@ -82,7 +87,8 @@ def __init__(self, address, configuration, frequency=1, initial_state=State.STAN
                 self.fault()
```

Comment:
The way it is coded here, if a `TimeoutError` is raised by `self.model.publish()` you go to a fault state but you still keep processing the loop. The next `if` statement will prevent the CSC from continuing publishing the `Fault` state but then you proceed to publish the telemetry values. Are you sure this is how you should proceed? I would think that, if `self.model.publish()` fails, you should not continue with the loop. 

Also, note that if there is an un-captured exception in some step, your telemetry loop will die and the CSC will probably not be informed at all. For instance, if the call to `self.model.publish()` raises anything other than `TimeoutError`, the `while` loop will die but the CSC will continue to run as if nothing had happened. 

Overall, I don't like the idea of your telemetry loop aways running. Except for the heartbeat, I think telemetry should only be published if the CSC is in a state above `Standby`. What I would suggest is that you keep that loop off and do the [`asyncio.ensure_future(self.telemetry())`](https://github.com/lsst-ts/ts_TunableLaser/blob/a9b8eeb80fa1bc1d721d8a5946933da432558f18/lsst/ts/laser/csc.py#L73) in `end_start`. Then, in `begin_standby` you make sure you cancel it. Then, make sure that if, for some reason, the loop dies, you send the CSC to fault state. Something like;

```
import traceback

...

    async def telemetry(self):
        try:
             while True:
                 ...
        except Exception as e:
            self.log.exception(e)
            self.evt_errorCode.set_put( errorCode=TELEMETRY_LOOP_DIED,
                                                           errorReport="Telemetry loop died for unspecified reason.",
                                                           traceback=traceback.format_exc() )
            self.fault()
```